api/src/main/java/uk/gov/dhsc/htbhf/claimant/repository/PaymentCycleRepository.java

```diff
@@ -1,12 +1,31 @@
 package uk.gov.dhsc.htbhf.claimant.repository;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
 
+import java.time.LocalDate;
+import java.util.List;
 import java.util.UUID;
 
 /**
  * JPA repository for {@link PaymentCycle}s.
  */
 public interface PaymentCycleRepository extends CrudRepository<PaymentCycle, UUID> {
+
+    /**
+     * Looks for all active (ACTIVE, PENDING_EXPIRY) claims whose latest PaymentCycle has an end date on or before cycleEndDate.
+     * @param cycleEndDate the date by which the cycle should end.
+     * @return a List of {@link ClosingPaymentCycle}s containing the claim id and cycle end date.
+     */
+    @Query(nativeQuery = true, value =
+            "SELECT Cast(claim_id as varchar) as claimIdString, cycle_end_date as cycleEndDateTimestamp FROM "
```

Comment:
We should create a plan to stress test this at some point if we're selecting all. See how many entries it takes to cause memory or performance issues.

Reply:
I do plan to test it locally with my 500k active claims

Reply:
We will also need another environment (Pre-Prod) in which we can load a lot of claims and leave it running.